test/stability/testMerge/testMerge.go

```diff
@@ -273,46 +587,115 @@ func generatePacketGroup2(pkt *packet.Packet, context flow.UserContext) {
 	}
 	ipv4 := pkt.GetIPv4()
 	udp := pkt.GetUDPForIPv4()
-	ipv4.SrcAddr = ipv4addr2
-	ipv4.HdrChecksum = packet.SwapBytesUint16(packet.CalculateIPv4Checksum(ipv4))
-	udp.DgramCksum = packet.SwapBytesUint16(packet.CalculateIPv4UDPChecksum(ipv4, udp, pkt.Data))
+	if isGroup1 {
+		ipv4.SrcAddr = ipv4addr1
+		fixMACAddrs1(pkt, context)
+	} else {
+		ipv4.SrcAddr = ipv4addr2
+		fixMACAddrs2(pkt, context)
+	}
 
-	fixMACAddrs2(pkt, context)
+	ptr := (*packetData)(pkt.Data)
+	ptr.N = uint(notCount)
 
 	// We do not consider the start time of the system in this test
 	if time.Since(progStart) >= T && atomic.LoadUint64(&recvPackets) < totalPackets {
-		atomic.AddUint64(&sentPacketsGroup2, 1)
+		if isGroup1 {
+			ptr.N = uint(baseGr1)
+			atomic.AddUint64(&sentPacketsGroup1, 1)
+		} else {
+			ptr.N = uint(baseGr2)
+			atomic.AddUint64(&sentPacketsGroup2, 1)
+		}
 	}
+
+	ipv4.HdrChecksum = packet.SwapBytesUint16(packet.CalculateIPv4Checksum(ipv4))
+	udp.DgramCksum = packet.SwapBytesUint16(packet.CalculateIPv4UDPChecksum(ipv4, udp, pkt.Data))
 }
 
 // Count and check packets in received flow
 func checkPackets(pkt *packet.Packet, context flow.UserContext) {
 	if time.Since(progStart) < T || stabilityCommon.ShouldBeSkipped(pkt) {
 		return
 	}
-	if atomic.AddUint64(&recvPackets, 1) >= totalPackets {
-		testDoneEvent.Signal()
+
+	if pkt.ParseData() < 0 {
+		println("ParseData returned negative value")
+		atomic.AddUint64(&brokenPackets, 1)
 		return
 	}
 
-	pkt.ParseData()
 	ipv4 := pkt.GetIPv4()
 	udp := pkt.GetUDPForIPv4()
 	recvIPv4Cksum := packet.SwapBytesUint16(packet.CalculateIPv4Checksum(ipv4))
 	recvUDPCksum := packet.SwapBytesUint16(packet.CalculateIPv4UDPChecksum(ipv4, udp, pkt.Data))
+	ptr := (*packetData)(pkt.Data)
+	payload := payloadVal(ptr.N)
+	if payload == notCount {
+		return
+	}
 
 	if recvIPv4Cksum != ipv4.HdrChecksum || recvUDPCksum != udp.DgramCksum {
 		// Packet is broken
 		atomic.AddUint64(&brokenPackets, 1)
 		return
 	}
 	if ipv4.SrcAddr == ipv4addr1 {
-		atomic.AddUint64(&recvPacketsGroup1, 1)
+		if payload == baseGr1 {
+			atomic.AddUint64(&recvPacketsGroup1, 1)
+		} else if payload == copiedGr1 {
+			atomic.AddUint64(&recvCopyPacketsGroup1, 1)
+		} else {
+			atomic.AddUint64(&brokenPackets, 1)
+		}
 	} else if ipv4.SrcAddr == ipv4addr2 {
-		atomic.AddUint64(&recvPacketsGroup2, 1)
+		if payload == baseGr2 {
+			atomic.AddUint64(&recvPacketsGroup2, 1)
+		} else if payload == copiedGr2 {
+			atomic.AddUint64(&recvCopyPacketsGroup2, 1)
+		} else {
+			atomic.AddUint64(&brokenPackets, 1)
+		}
 	} else {
 		println("Packet Ipv4 src addr does not match addr1 or addr2")
 		println("TEST FAILED")
 		atomic.StoreInt32(&passed, 0)
 	}
+	if atomic.AddUint64(&recvPackets, 1) >= totalPackets {
+		testDoneEvent.Signal()
+		return
+	}
+}
+
+// Count packets in copied flow
+func countCopiedPackets(pkt *packet.Packet, context flow.UserContext) {
```

Comment:
We don't need this function here.
You set 0 to passed at different machine - it won't make any effect. Also you shouldn't check copy here. Any copy problems will be seen in Copy test.

Reply:
Also you count copied packets at separate machine which won't produce any result. -> remove this function.

Reply:
I need handler, cause change packet's payload in it.
Counted packets are used in gotest.

Reply:
please, rename this function to packetChaning or something like this. And use it everywhere, not only after copies if you want.
Remove copy counting stuff. It is not about merge. If you would like to count packets after copy you should write separate test which will be called only with gotest

Reply:
It is a very bad practice to have tests which can PASS (after using test system), but test something ELSE when using gotest. What???